pypeit/pypeit.py

```diff
@@ -566,84 +575,59 @@ def extract_one(self, frames, det, bg_frames=[], std_outfile=None):
         # For QA on crash
         msgs.sciexp = self.sciImg
 
-        # Object finding, first pass on frame without sky subtraction
+        # Instantiate Reduce object
         self.maskslits = self.caliBrate.tslits_dict['maskslits'].copy()
-
+        # Required for pypeline specific object
+        # TODO -- caliBrate should be replaced by the ~3 primary Objects needed
+        #   once we have the data models in place.
         self.redux = reduce.instantiate_me(self.sciImg, self.spectrograph,
-                                           self.caliBrate.tslits_dict, self.par,
-                                           self.caliBrate.tilts_dict['tilts'],
+                                           self.par, self.caliBrate,
```

Comment:
In my opinion, the parset should perhaps be optional since there are sensible defaults that you can pull from spectrograph.

Reply:
That is a major code philosophy change that I'm not walking down at this stage.
Current philosophy is that `par` is always generated at the start and immutable.